File: src/services/vehicleService.js
import { api } from "./api";

export async function listVehicles(params = {}) {
  return api.get("/vehicles", { params }).then((r) => r.data);
}

export async function getVehicleById(id) {
  return api.get(`/vehicles/${id}`).then((r) => r.data);
}

export async function createVehicle(payload) {
  return api.post("/vehicles", payload).then((r) => r.data);
}

export async function updateVehicle(id, payload) {
  return api.put(`/vehicles/${id}`, payload).then((r) => r.data);
}

export async function deleteVehicle(id) {
  return api.delete(`/vehicles/${id}`).then((r) => r.data);
}

File: backend/src/main/java/com/falconenergy/service/VehicleService.java
package com.falconenergy.service;

import com.falconenergy.dto.VehicleRequest;
import com.falconenergy.dto.VehicleResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface VehicleService {
    VehicleResponse createVehicle(VehicleRequest request);
    VehicleResponse getVehicleById(Long id);
    VehicleResponse updateVehicle(Long id, VehicleRequest request);
    void deleteVehicle(Long id);
    Page<VehicleResponse> getAllVehicles(String search, String status, Pageable pageable);
}
